job-server/src/spark.jobserver/JobServerNamedRdds.scala

```diff
@@ -87,7 +87,11 @@ class JobServerNamedRdds(val rddManager: ActorRef) extends NamedRdds {
       "forceComputation implies storageLevel != NONE")
     val rdd = rddGen
     rdd.setName(name)
-    if (storageLevel != StorageLevel.NONE) rdd.persist(storageLevel)
+    val level = rdd.getStorageLevel match {
+      case StorageLevel.NONE => storageLevel
+      case _ => rdd.getStorageLevel
```

Comment:
I would just combine the match and the next persist line.  Also no need to call `getStorageLevel` twice.
```
rdd.getStorageLevel match {
    case StorageLevel.NONE => rdd.persist(storageLevel)
    case alreadySet                => rdd.persist(alreadySet)
}
```